.dotnet.azure/src/Generated/ServiceVersion.cs

```diff
@@ -0,0 +1,53 @@
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.AI.OpenAI
+{
+    /// <summary> Known service API versions for Azure OpenAI. </summary>
+    public readonly partial struct ServiceVersion : IEquatable<ServiceVersion>
+    {
+        private readonly string _value;
+
+        /// <summary> Initializes a new instance of <see cref="ServiceVersion"/>. </summary>
+        /// <exception cref="ArgumentNullException"> <paramref name="value"/> is null. </exception>
+        public ServiceVersion(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string V20240401PreviewValue = "2024-04-01-preview";
+        private const string V20240501PreviewValue = "2024-05-01-preview";
+        private const string V20240601Value = "2024-06-01";
+        private const string V20240701PreviewValue = "2024-07-01-preview";
+        private const string V20240801PreviewValue = "2024-08-01-preview";
```

Comment:
Should this only target the latest preview?

Reply:
Historically, we haven't removed service API labels from the spec and SDK lists until those labels themselves were removed on the service side. It's very conceivable that a customer might want to do (when possible and applicable, given model compatibility) an A/B comparison during e.g. acceptance testing,

Reply:
I think the fact that we haven't is a mistake on our side because my understanding is that this is how we do it in other Azure libraries. My understanding is that the only versions that remain forever are GA versions. For preview versions, we should only keep the latest preview that comes after the latest GA.